org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Replay/ImportReplay.java

```diff
@@ -541,6 +541,36 @@ public final void testParseSymbologyColor()
       assertEquals("sym type correct", "Hidar", label2.getSymbolType());
       assertEquals("color correct", DebriefColors.PURPLE, label2.getColor());
     }
+    
+    
+    public final void testPasteRep() {
+      String textToPaste =
+";LINE: @B 20 50 0 N 21 10 0 W 22 0 0 N 21 10 0 W test line\r\n"+
+";VECTOR: @C 21.6 12 0 N 21.5 11 0 W 5000 270 test vector\r\n"+
+";CIRCLE: @D    21.8 0 0 N 21.0 0 0 W 2000 test circle\r\n"+
+";TEXT: @E 21.7 0 0 N 21.5 0 0 W test text\r\n"+ 
+";TEXT: WB 21.72 0 0 N 21.52 0 0 W wreck symbol\r\n"+ 
+";TEXT: CA[LAYER=Special_Layer] 21.42 0 0 N 21.88 0 0 W Other layer\r\n"+
+";TEXT: CA[LAYER=Other_Special_Layer] 21.22 0 0 N 21.88 0 0 W Other layer 3\r\n"+
+";ELLIPSE: @F 951212 060200 21.8 0 0 N 21.5 0 0 W 45.0 5000 3000 test ellipse\r\n"+
+";POLY: @GA30 21.9 0 0 N 21.5 0 0 W 22 0 0 N 21.8 0 0 W 22.1 0 0 N 21.5 0 0 W test poly\r\n"+
+";POLYLINE: @C 21.1 0 0 N 21.5 0 0 W 21.2 0 0 N 21.8 0 0 W 21.3 0 0 N 21.5 0 0 W test polyline\r\n"+
+";NARRATIVE:  951212 050200 NEL_STYLE comment 3\r\n";
+      ImportReplay testImporter = new ImportReplay();
+      Layers tmpLayers = new Layers();
+      Layers dest = new Layers();
+      testImporter.setLayers(tmpLayers);
+      testImporter.importThis(textToPaste);
+      testImporter.injectContent(tmpLayers, dest, true);
+      assertTrue(((Layer)tmpLayers.findLayer(ANNOTATION_LAYER)).elements().hasMoreElements());
```

Comment:
@ayeshare - could you also check that the layer contains the correct number of elements?